app/resources/services/integrations/coil.js

```diff
@@ -0,0 +1,183 @@
+const config = require('config')
+
+const btoa = require('btoa')
+
+const passport = require('passport')
+const OAuth2Strategy = require('passport-oauth').OAuth2Strategy
+const InternalOAuthError = require('passport-oauth2').InternalOAuthError
+const axios = require('axios')
+const { User } = require('../../../db/models')
+const logger = require('../../../../lib/logger.js')()
+
+const {
+  findUser,
+  addOrUpdateByProviderName,
+  syncAccountStatus
+} = require('./helpers')
+
+const authToken = btoa(
+  process.env.COIL_CLIENT_ID +
+    ':' +
+    encodeURIComponent(process.env.COIL_CLIENT_SECRET)
+)
+const coilStrategy = new OAuth2Strategy(
```

Comment:
Similar to how we handle Patreon, let's make a way to exit out of running this gracefully if neither `COIL_CLIENT_ID` or `COIL_CLIENT_SECRET` is found. Right now, the non-existence of this env var will crash the application in the Cypress E2E tests (and it will prevent anyone from running Streetmix locally without a Coil ID, which we don't want).